### ISSUE TITLE
Error using Github package : libcurl.so.4: version `CURL_OPENSSL_3' not found

### ISSUE THREAD

**Issue by kunjeshn** [atom/github#1680]
### Description

Using github package that comes by default on atom throws error.

### Steps to Reproduce

1. Running Atom 1.30.0 on Ubuntu 18 with all updates installed.
2. open any git repo
3. click on fetch on the github menu on lower right.

**Expected behavior:**

To fetch from the remote repo.

**Actual behavior:**
Throws an error as below.

/snap/atom/192/usr/share/atom/resources/app.asar.unpacked/node_modules/dugite/git/libexec/git-core/git-remote-http: /usr/lib/x86_64-linux-gnu/libcurl.so.4: version `CURL_OPENSSL_3' not found (required by /snap/atom/192/usr/share/atom/resources/app.asar.unpacked/node_modules/dugite/git/libexec/git-core/git-remote-http)
/snap/atom/192/usr/share/atom/resources/app.asar.unpacked/node_modules/dugite/git/libexec/git-core/git-remote-http: /snap/core/current/lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.25' not found (required by /usr/lib/x86_64-linux-gnu/libgnutls.so.30)
/snap/atom/192/usr/share/atom/resources/app.asar.unpacked/node_modules/dugite/git/libexec/git-core/git-remote-http: /snap/core/current/lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.26' not found (required by /usr/lib/x86_64-linux-gnu/libp11-kit.so.0)

**Reproduces how often:**

Atom throws error everytime any github package options is used.

### Versions

Atom ver 1.30.0

**Comment by rsese** [comment]
Thanks for the report!  It looks like you're using the Snap install which is unofficial - can you confirm if you have the same issue using one of the official installation methods?

https://flight-manual.atom.io/getting-started/sections/installing-atom/#installing-atom-on-linux

**Comment by kunjeshn** [comment]
Always welcome. Installing from the package repository seems to solve the error partially. But i still not able to fetch pull or push. error shown is as below. 
![screenshot from 2018-09-06 10-32-14](https://user-images.githubusercontent.com/25734767/45136086-34a70b80-b1c0-11e8-8745-5bb0c962c135.png)


/usr/share/atom/resources/app.asar.unpacked/node_modules/dugite/git/libexec/git-core/git-remote-http: /usr/lib/x86_64-linux-gnu/libcurl.so.4: version `CURL_OPENSSL_3' not found (required by /usr/share/atom/resources/app.asar.unpacked/node_modules/dugite/git/libexec/git-core/git-remote-http)

**Comment by rsese** [comment]
Ahhh sorry, I see this was already reported so I'll close in favor of https://github.com/atom/github/issues/848 and you can subscribe there for updates if you'd like.

**Comment by MensSana** [comment]
Fixed it by doing this on Ubuntu 18.04 LTS:

`apt install libcurl3 libcurl-openssl1.0-dev`

**Comment by dmknght** [comment]
> Fixed it by doing this on Ubuntu 18.04 LTS:
> 
> `apt install libcurl3 libcurl-openssl1.0-dev`

```
The following packages will be REMOVED:
  apache2 apache2-bin clamav clamtk curl dirb gnome-user-share
  libapache2-mod-dnssd libapache2-mod-php7.1 libapache2-mod-php7.2 libcurl4
  libcurl4:i386 php7.1 playonlinux
The following NEW packages will be installed:
  libcurl3 php7.2-fpm
```
Nah no! That is not good for me (Debian 10)!
I downgraded atom to 1.30.0 and it works okay. Waiting for the fix in next version.
`apt install atom=1.30.0`

**Comment by burningTyger** [comment]
is this a degit issue? I'm seeing this with the current degit package in my electron app, too.

**Comment by rsese** [comment]
Also see a workaround mentioned in https://github.com/atom/github/issues/848#issuecomment-429575264 and feel free to subscribe to that issue for updates.

**Comment by kunjeshn** [comment]
Hey. error resolved by re-enabling git-plus community package. let me know if this deals better than the mentioned workaround in [#848 (comment) ](https://github.com/atom/github/issues/848#issuecomment-429575264)..

Atom 1.31.2 x64
https://atom.io/packages/git-plus - ver 8.1.0

**Comment by NikitaPO** [comment]
Thank you for your help!

> Fixed it by doing this on Ubuntu 18.04 LTS:
> 
> `apt install libcurl3 libcurl-openssl1.0-dev`

**Comment by Spitfire1900** [comment]
The workaround no longer works in Ubuntu 18.10 as libcurl3 has been replaced with libcurl3-gnutls and libcurl-openssl1.0-dev (which depends on libcurl3) has been replaced with libcurl4-openssl-dev (depends libcurl4); /usr/lib/x86_64-linux-gnu/libcurl.so.4 is now owned by package libcurl4.

Error:
```/usr/share/atom/resources/app.asar.unpacked/node_modules/dugite/git/libexec/git-core/git-remote-https: /usr/lib/x86_64-linux-gnu/libcurl.so.4: version `CURL_OPENSSL_3' not found (required by /usr/share/atom/resources/app.asar.unpacked/node_modules/dugite/git/libexec/git-core/git-remote-https)```

atom --version:
```Atom    : 1.32.0
Electron: 2.0.9
Chrome  : 61.0.3163.100
Node    : 8.9.3
```
apm --version
```apm  2.1.2
npm  6.2.0
node 8.9.3 x64
atom 1.32.0
python 2.7.15+
git 2.19.1
```

apm list | grep github
```├── github@0.20.1```


lsb_release -a:
```No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 18.10
Release:        18.10
Codename:       cosmic
```

dpkg -l | grep libcurl:
```dpkg -l | grep libcurl
ii  libcurl3-gnutls:amd64                                       7.61.0-1ubuntu2                             amd64        easy-to-use client-side URL transfer library (GnuTLS flavour)
ii  libcurl4:amd64                                              7.61.0-1ubuntu2                             amd64        easy-to-use client-side URL transfer library (OpenSSL flavour)
ii  python3-pycurl                                              7.43.0.1-0.2build1                          amd64        Python bindings to libcurl (Python 3)
```
apt-file find /usr/lib/x86_64-linux-gnu/libcurl.so.4
```libcurl4: /usr/lib/x86_64-linux-gnu/libcurl.so.4
libcurl4: /usr/lib/x86_64-linux-gnu/libcurl.so.4.5.0
```

**Package needs to search for libcurl-gnutls.so.4 or upgrade to libcurl4**
apt-file list libcurl3-gnutls
```libcurl3-gnutls: /usr/lib/x86_64-linux-gnu/libcurl-gnutls.so.3
libcurl3-gnutls: /usr/lib/x86_64-linux-gnu/libcurl-gnutls.so.4
libcurl3-gnutls: /usr/lib/x86_64-linux-gnu/libcurl-gnutls.so.4.5.0
libcurl3-gnutls: /usr/share/doc/libcurl3-gnutls/NEWS.Debian.gz
libcurl3-gnutls: /usr/share/doc/libcurl3-gnutls/changelog.Debian.gz
libcurl3-gnutls: /usr/share/doc/libcurl3-gnutls/copyright
libcurl3-gnutls: /usr/share/lintian/overrides/libcurl3-gnutls
```

apt-cache show libcurl4-openssl-dev | grep Depends
```Depends: libcurl4 (= 7.61.0-1ubuntu2)
```

**Comment by discoverdata** [comment]
This removed my r-base-core files. Apparently, libcurl3 and libcurl4 cant be installed together. I found a solution here https://dev.to/jake/using-libcurl3-and-libcurl4-on-ubuntu-1804-bionic-184g_

**Comment by discoverdata** [comment]
__

> Fixed it by doing this on Ubuntu 18.04 LTS:
> 
> `apt install libcurl3 libcurl-openssl1.0-dev`

This removed my r-base-core files. 
dpkg: libcurl4:amd64: dependency problems, but removing anyway as you requested:
 r-base-core depends on libcurl4 (>= 7.28.0).
For people who want both libcurl3 and libcurl4
Apparently, libcurl3 and libcurl4 cant be installed together. I found a solution here https://dev.to/jake/using-libcurl3-and-libcurl4-on-ubuntu-1804-bionic-184g_

**Comment by discoverdata** [comment]
> Fixed it by doing this on Ubuntu 18.04 LTS:
> 
> `apt install libcurl3 libcurl-openssl1.0-dev`


This removed my r-base-core files.
dpkg: libcurl4:amd64: dependency problems, but removing anyway as you requested:
r-base-core depends on libcurl4 (>= 7.28.0).
Apparently, libcurl3 and libcurl4 cant be installed together. For people who want both libcurl3 and libcurl4, I found a solution here:
https://askubuntu.com/questions/1061431/how-to-have-both-libcurl3-and-libcurl4-installed-at-same-time

**Comment by Arcanemagus** [comment]
[As stated](https://github.com/atom/github/issues/848#issuecomment-435291468) in the issue this is marked as a duplicate of, this has been fixed since Atom v1.32.1 / v1.33.0-beta1.

If you are still experiencing this problem with a current version of Atom please file a **new** issue, thanks!
